plantio/serializers/aplicacao.py

```diff
@@ -2,32 +2,58 @@
 
 from plantio.models import AplicacaoAgrotoxico
 
+from core.consts.agrotoxicos import AplicacaoEstados
+
 from rest_framework import serializers
 
 
-# TODO: As linhas comentadas fazem referência a implementação da issue 152
 class AplicacaoAgrotoxicoSerializer(serializers.ModelSerializer):
-    # fotoAgrotoxico = serializers
-
     class Meta:
         model = AplicacaoAgrotoxico
         fields = (
-            # 'idAgrotoxico', 'dataAplicacao', 'fotoAgrotoxico', 'dosagemAplicacao'
-            'plantio', 'agrotoxico', 'dataAplicacao', 'dosagemAplicacao'
+            'plantio', 'agrotoxico', 'dataAplicacao', 'dosagemAplicacao', 'fotoAgrotoxico', 'estadoAnalise'
         )
+        read_only_fields = ('estadoAnalise',)
+        extra_kwargs = {
+            'agrotoxico': {'required': False},
+            'fotoAgrotoxico': {'required': False}
+        }
+
+    def create(self, validated_data):
+        estadoAnalise = AplicacaoEstados.SUCESSO
+
+        if validated_data.get("fotoAgrotoxico"):
+            estadoAnalise = AplicacaoEstados.ANALISE
+
+        return AplicacaoAgrotoxico.objects.create(estadoAnalise=estadoAnalise, **validated_data)
 
     def validate_dataAplicacao(self, dataAplicacao):
         if dataAplicacao > date.today():
             raise serializers.ValidationError('Data de aplicação no futuro.')
         return dataAplicacao
 
     def validate(self, data):
-        if data.get('agrotoxico'):
-            query = AplicacaoAgrotoxico.objects.filter(
-                plantio=data['plantio'], agrotoxico=data['agrotoxico'], dataAplicacao=data['dataAplicacao'])
 
-            if query.exists():
-                raise serializers.ValidationError(
-                    f'Esse agrotóxico já foi aplicado nessa plantação na data {data["dataAplicacao"]}.')
+        self._validate_agrotoxico_xor_fotoAgrotoxico(data.get('agrotoxico'), data.get('fotoAgrotoxico'))
+        self._validate_agrotoxico(data.get('agrotoxico'), data['plantio'], data['dataAplicacao'])
 
         return data
+
+    def _validate_agrotoxico_xor_fotoAgrotoxico(self, agrotoxico, fotoAgrotoxico):
+        if (agrotoxico is not None) ^ (fotoAgrotoxico is not None):
+            return
+
+        raise serializers.ValidationError(
+            'É necessário enviar ou o agrotóxico ou a foto do agrotóxico.'
+        )
```

Comment:
Não ficaria mais enxuto fazer um not xor ? Porque ai vc pode só fazer o raise sem precisar de um "return void" 😷🤒🤕

Reply:
A ideia é essa mesma, desse jeito tem um nivel de identação a menos. Além disso, entender um xor é bem mais facil que um not xor, ou seja, traria uma complexidade a mais desnecessaria